### ISSUE TITLE
Some more failing CVE's

### ISSUE THREAD

**Issue by StephenBrown2** [evitalis/cvechk#7]
Possibly multiple issues here...

CVE-2016-6210 (OpenSSH timing attack) - 6 good - 7 good
CVE-2014-0160 (Heartbleed) - 6 good - 7 bad
CVE-2014-6271 (ShellShock) - 6 good - 7 good
CVE-2016-2183 (Sweet32)    - 6 bad  - 7 bad
CVE-2016-5195 (DirtyCOW)   - 6 good - 7 good

e.g. I'm getting this error for the 'bad' ones:
```
cvechk_1  | 172.20.0.1 - - [20/Mar/2017 14:17:00] "POST /results HTTP/1.1" 500 -
cvechk_1  | [2017-03-20 14:17:10,955] ERROR in app: Exception on /results [POST]
cvechk_1  | Traceback (most recent call last):
cvechk_1  |   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1982, in wsgi_app
cvechk_1  |     response = self.full_dispatch_request()
cvechk_1  |   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1614, in full_dispatch_request
cvechk_1  |     rv = self.handle_user_exception(e)
cvechk_1  |   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1517, in handle_user_exception
cvechk_1  |     reraise(exc_type, exc_value, tb)
cvechk_1  |   File "/usr/local/lib/python3.6/site-packages/flask/_compat.py", line 33, in reraise
cvechk_1  |     raise value
cvechk_1  |   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1612, in full_dispatch_request
cvechk_1  |     rv = self.dispatch_request()
cvechk_1  |   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1598, in dispatch_request
cvechk_1  |     return self.view_functions[rule.endpoint](**req.view_args)
cvechk_1  |   File "/usr/src/app/cvechk/views.py", line 23, in results
cvechk_1  |     cachedata = redis_get_data(oschoice, cves)
cvechk_1  |   File "/usr/src/app/cvechk/utils.py", line 34, in redis_get_data
cvechk_1  |     cvedata[cve] = mod_rhel.rh_get_pkgs(os, cve)
cvechk_1  |   File "/usr/src/app/cvechk/osmods/mod_rhel.py", line 57, in rh_get_pkgs
cvechk_1  |     redis_set_data('cvechk:{}:{}'.format(os, cve), cvedata)
cvechk_1  |   File "/usr/src/app/cvechk/utils.py", line 45, in redis_set_data
cvechk_1  |     redis_conn.hmset(key, cvedata)
cvechk_1  |   File "/usr/local/lib/python3.6/site-packages/redis/client.py", line 1901, in hmset
cvechk_1  |     raise DataError("'hmset' with 'mapping' of length 0")
cvechk_1  | redis.exceptions.DataError: 'hmset' with 'mapping' of length 0
```

It seems that sometimes there are multiple packages affected or not, which is throwing things off.

e.g. for CVE-2016-2183:
```
        {
            "product_name": "Red Hat Enterprise Linux 6",
            "fix_state": "Affected",
            "package_name": "openssl",
            "cpe": "cpe:/o:redhat:enterprise_linux:6"
        },
        {
            "product_name": "Red Hat Enterprise Linux 6",
            "fix_state": "Not affected",
            "package_name": "gnutls",
            "cpe": "cpe:/o:redhat:enterprise_linux:6"
        },
        {
            "product_name": "Red Hat Enterprise Linux 6",
            "fix_state": "Will not fix",
            "package_name": "nss",
            "cpe": "cpe:/o:redhat:enterprise_linux:6"
        },
```

And for heartbleed (CVE-2014-0160), RHEL 6 is affected, but RHEL 7 is not.

**Comment by evitalis** [comment]
Crashed solved with commit 2db726b14d3f, but output issues with the following CVEs.

CVE-2016-0160 (RHEL 7)
CVE-2016-2183 (RHEL 6, RHEL 7)

**Comment by evitalis** [comment]
This should be able to get narrowed down with the newest logging commits.

Will verify the logs after a few days to determine exact errors being returned.

**Comment by evitalis** [comment]
Resolved by 8eec58a9055047
